tests/tpch/generate-data.py

```diff
@@ -1,79 +1,181 @@
+import functools
 import pathlib
 import tempfile
+import warnings
 
+import boto3
 import botocore.session
 import click
+import coiled
 import dask
 import duckdb
 import psutil
+import pyarrow.compute as pc
+from dask.distributed import LocalCluster
+
+REGION = None
 
 
 def generate(
-    scale: int = 10, partition_size: str = "50 MiB", path: str = "./tpch-data"
+    scale: int = 10,
+    partition_size: str = "128 MiB",
+    path: str = "./tpch-data",
+    relaxed_schema: bool = False,
 ):
     if str(path).startswith("s3"):
         path += "/" if not path.endswith("/") else ""
-        path += f"scale-{scale}/"
+        path += f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        use_coiled = True
+        global REGION
+        REGION = get_bucket_region(path)
     else:
-        path = pathlib.Path(path) / f"scale-{scale}/"
+        path = (
+            pathlib.Path(path)
+            / f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        )
         path.mkdir(parents=True, exist_ok=True)
+        use_coiled = False
 
     print(f"Scale: {scale}, Path: {path}, Partition Size: {partition_size}")
-
-    con = duckdb.connect()
-    con.install_extension("tpch")
-    con.load_extension("tpch")
-
-    session = botocore.session.Session()
-    creds = session.get_credentials()
-    con.install_extension("httpfs")
-    con.load_extension("httpfs")
-    con.sql(
-        f"""
-        SET memory_limit='{psutil.virtual_memory().available // 2**30 }G';
-        SET s3_region='us-east-2';
-        SET s3_access_key_id='{creds.access_key}';
-        SET s3_secret_access_key='{creds.secret_key}';
-        SET s3_session_token='{creds.token}';
-        """
+    kwargs = dict(
+        scale=scale,
+        path=path,
+        relaxed_schema=relaxed_schema,
+        partition_size=partition_size,
     )
 
-    print("Generating TPC-H data")
-    con.sql(f"call dbgen(sf={scale})")
-    print("Finished generating data, exporting...")
+    if use_coiled:
+        with coiled.Cluster(
+            n_workers=10,
+            # workload is best with 1vCPU and ~3-4GiB memory
+            worker_vm_types=["m7a.medium", "m3.medium"],
+            worker_options={"nthreads": 1},
+            region=REGION,
+        ) as cluster:
+            cluster.adapt(minimum=1, maximum=350)
+            _generate_via_cluster(cluster, **kwargs)
+    else:
+        with LocalCluster(threads_per_worker=1) as cluster:
+            _generate_via_cluster(cluster, **kwargs)
```

Comment:
Sorry to drag on review, but I'm curious, why put LocalCluster in the loop here?  

Personally I like it when I have a local option that doesn't use Dask, because then it's easy to do things like profile, add breakpoints, etc..

Reply:
Ah, I can see that point. I placed it here to mirror the workflow which happens when using Coiled. Happy to remove it though in this PR (see you've approved already though)

Reply:
Removing is good.  Feel free to merge after removing (or I can remove after you merge).

Reply:
Done https://github.com/coiled/benchmarks/pull/1131/commits/0e4faaca83c3af79234d7a728df83e7b2c8c4e25